crates/astria-bridge-withdrawer/src/bridge_withdrawer/submitter/mod.rs

```diff
@@ -274,22 +271,22 @@ async fn submit_tx(
     res
 }
 
-// TODO(https://github.com/astriaorg/astria/issues/1274): deduplicate here and in crate::bridge_withdrawer::startup
-async fn get_pending_nonce(
+pub(crate) async fn get_pending_nonce(
     client: sequencer_service_client::SequencerServiceClient<Channel>,
     address: Address,
     state: Arc<State>,
-    // metrics: &'static Metrics,
+    metrics: Option<&'static Metrics>,
```

Comment:
I think we should be able to make this non-optional.  It will mean threading the `Metrics` through a few more places like the `Builder` and `Startup` types, but it shouldn't be a big lift I think.

The benefit is that we get metrics during the startup phase (which I think is the only time this is `None`) and that can help expose the problem if we we're unable to start due to repeatedly failing to fetch the nonce.  It's also a little more efficient for the normal flow after startup (which will be the vast majority of calls here usually) since the code isn't branching on an option which is eternally `Some` after startup.